cleverhans/feat_advs.py

```diff
@@ -0,0 +1,265 @@
+        :param clip_min: (optional float) Minimum input component value
+        :param clip_max: (optional float) Maximum input component value
+        """
+        # Save attack-specific parameters
+
+        self.layer = layer
+        self.eps = eps
+        self.ord = ord
+        self.clip_min = clip_min
+        self.clip_max = clip_max
+
+        # Check if order of the norm is acceptable given current implementation
+        if self.layer not in self.model.get_layer_names():
+            raise ValueError("Layer name does not exist in the model layers.")
+        if self.ord not in [np.inf, int(1), int(2)]:
+            raise ValueError("Norm order must be either np.inf, 1, or 2.")
+        if self.back == 'th' and self.ord != np.inf:
+            raise NotImplementedError("The only FastGradientMethod norm "
+                                      "implemented for Theano is np.inf.")
+        return True
+
+
+class FastIterativeFeatureAdversaries(Attack):
+
+    """
+    This is similar to Basic Iterative Method (Kurakin et al. 2016) but
+    applied to the internal representations.
+    """
+
+    def __init__(self, model, back='tf', sess=None):
+        """
+        Create a FeatureAdversaries instance.
+        """
+        super(FastIterativeFeatureAdversaries, self).__init__(model, back,
+                                                              sess)
+        self.feedable_kwargs = {'eps': np.float32,
+                                'eps_iter': np.float32,
+                                'clip_min': np.float32,
+                                'clip_max': np.float32,
+                                'layer': str}
+        self.structural_kwargs = ['ord', 'nb_iter']
+
+        assert isinstance(self.model, Model)
+
+    def generate(self, x, g, **kwargs):
+        """
+        Generate symbolic graph for adversarial examples and return.
+        :param x: The model's symbolic inputs.
+        :param g: The target's symbolic representation.
+        :param eps: (required float) maximum distortion of adversarial example
+                    compared to original input
+        :param eps_iter: (required float) step size for each attack iteration
+        :param nb_iter: (required int) Number of attack iterations.
+        :param y: (optional) A tensor with the model labels.
+        :param y_target: (optional) A tensor with the labels to target. Leave
+                         y_target=None if y is also set. Labels should be
+                         one-hot-encoded.
+        :param ord: (optional) Order of the norm (mimics Numpy).
+                    Possible values: np.inf, 1 or 2.
+        :param clip_min: (optional float) Minimum input component value
+        :param clip_max: (optional float) Maximum input component value
+        """
+        import tensorflow as tf
+
+        # Parse and save attack-specific parameters
+        assert self.parse_params(**kwargs)
+
+        def clip_eta(eta):
```

Comment:
I think you can use `clip_eta` function defined in `utils_tf.py` [here](https://github.com/tensorflow/cleverhans/blob/5114ebd6b54c4197819042a7d716b2ac4e8285a0/cleverhans/utils_tf.py#L396-L421).

Reply:
Fixed in b73717bdc1206508d1e425fd84a5b1298ba59705.